File: Codeforces/CPP/479A.cpp
/*
Name: Expression
Link: https://codeforces.com/problemset/problem/479/A
Categories: math, max
*/

#include <bits/stdc++.h>
using namespace std;

int main()
{
    // inputs as integer
    int a, b, c, r1, r2, r3, r4, r5;
    cin >> a >> b >> c;

    // g1 vector is for dynamic array
    vector<unsigned int> array;

    // all possibilities
    r1 = a + b * c;
    r2 = a * (b + c);
    r3 = a * b * c;
    r4 = (a + b) * c;
    r5 = a + b + c;

    // inserting multiple elements to dynamic array
    array.insert(array.end(), {r1, r2, r3, r4, r5});

    // sorting an array
    sort(array.begin(), array.end());
    cout << array[0];
    return 0;
}

File: Codeforces/Python/479A.py
'''
Name: Expression
Link: https://codeforces.com/problemset/problem/479/A
Categories: math, max
'''

# inputs as integer
a = int(input())
b = int(input())
c = int(input())

# all possibilities
c1 = a + b * c
c2 = a * (b + c)
c3 = a * b * c
c4 = (a + b) * c
c5 = a + b + c

# max is used find maximum of given values
print(max(c1, c2, c3, c4, c5))
